src/Provider/AbstractProvider.php

```diff
@@ -98,6 +99,15 @@ abstract class AbstractProvider
      */
     protected $optionProvider;
 
+    /**
+     * If a response body cannot be parsed, a value true of this flag will allow
+     * the response parser to throw a ResponseParsingException containing
+     * the response and the body.
+     *
+     * @var bool
+     */
+    protected $mayThrowResponseParsingException = false;
```

Comment:
I don't like the `$mayThrowResponseParsingException` property. I think it introduces complexity that's not needed. I think there's a better way to preserve BC. See below.

Reply:
Agree.